LibreNMS/Modules/Isis.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Isis.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @link       http://librenms.org
+ * @copyright  2021 Otto Reinikainen
+ * @author     Otto Reinikainen <otto@ottorei.fi>
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Models\Device;
+use App\Models\IsisAdjacency;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\OS\Junos;
+use LibreNMS\Util\IP;
+
+class Isis implements Module
+{
+    /**
+     * Discover this module. Heavier processes can be run here
+     * Run infrequently (default 4 times a day)
+     *
+     * @param OS $os
+     */
+    public function discover(OS $os)
+    {
+        // not implemented
+    }
+
+    /**
+     * Poll data for this module and update the DB / RRD.
+     * Try to keep this efficient and only run if discovery has indicated there is a reason to run.
+     * Run frequently (default every 5 minutes)
+     *
+     * @param OS $os
+     */
+    public function poll(OS $os)
+    {
+        // Get device objects
+        $device_array = $os->getDeviceArray();
+        $device = $os->getDevice();
+        $device_id = $os->getDeviceId();
+
+        // Poll all ISIS enabled circuits from the device
+        $circuits_poll = snmpwalk_group($device_array, 'ISIS-MIB::isisCirc', 'ISIS-MIB');
+
+        // Poll all available adjacencies
+        $adjacencies_poll = snmpwalk_group($device_array, 'ISIS-MIB::isisISAdj', 'ISIS-MIB');
+        $adjacencies = collect();
+        $isis_data = [];
+
+        // Translate system state codes into meaningful strings
+        $isis_codes['1'] = 'L1';
```

Comment:
Please initialize $isis_codes.  The best way is to just set it to a fully formed array.

Reply:
Initialized